### ISSUE TITLE
Incorrect front padding result with `ttnn.pad` when padding length < 16

### ISSUE THREAD

**Issue by jerrysky3** [tenstorrent/pytorch2.0_ttnn#192]
When doing front padding with `ttnn.pad` in `ROW_MAJOR` layout, the padded result is incorrect when front padding length < 16

### Reproduce

Run the example below:

```py
import torch
import ttnn


def main(device):
    ptensor = torch.ones(32, dtype=torch.bfloat16)
    tensor = ttnn.from_torch(ptensor, layout=ttnn.ROW_MAJOR_LAYOUT, device=device)
    tensor = ttnn.pad(tensor, padding=[(2, 0)], value=0.5)
    print(tensor)


if __name__ == "__main__":
    device_id = 0
    device = ttnn.open_device(device_id=device_id)
    main(device)
```

Expected output:

```py
ttnn.Tensor([ 0.50000,  0.50000,  ...,  1.00000,  1.00000], shape=Shape([34]), dtype=DataType::BFLOAT16, layout=Layout::ROW_MAJOR)
```

Actual output:

```py
ttnn.Tensor([ 1.00000,  1.00000,  ...,  0.50000,  0.50000], shape=Shape([34]), dtype=DataType::BFLOAT16, layout=Layout::ROW_MAJOR)
```

However, if we change the the front padding length to 16 `[(16, 0)]`, then it pads correctly

```py
ttnn.Tensor([ 0.50000,  0.50000,  ...,  1.00000,  1.00000], shape=Shape([48]), dtype=DataType::BFLOAT16, layout=Layout::ROW_MAJOR)
```

**Comment by swimdi** [comment]
These input varation are appeared in models and is related to this constraint
```
        ["Tensor<[1, 144, 56, 56]> self = ?", "List[int] pad = [1, 2, 1, 2]", "number value = 0.0"],
        ["Tensor<[1, 672, 14, 14]> self = ?", "List[int] pad = [1, 2, 1, 2]", "number value = 0.0"],
        ["Tensor<[1, 144, 60, 60]> self = ?", "List[int] pad = [1, 2, 1, 2]", "number value = 0.0"],
        ["Tensor<[1, 672, 15, 15]> self = ?", "List[int] pad = [2, 2, 2, 2]", "number value = 0.0"],
        ["Tensor<[1, 144, 65, 65]> self = ?", "List[int] pad = [2, 2, 2, 2]", "number value = 0.0"],
        ["Tensor<[1, 288, 33, 33]> self = ?", "List[int] pad = [1, 1, 1, 1]", "number value = 0.0"],
        ["Tensor<[1, 720, 17, 17]> self = ?", "List[int] pad = [2, 2, 2, 2]", "number value = 0.0"],
        ["Tensor<[1, 192, 75, 75]> self = ?", "List[int] pad = [2, 2, 2, 2]", "number value = 0.0"],
        ["Tensor<[1, 816, 19, 19]> self = ?", "List[int] pad = [2, 2, 2, 2]", "number value = 0.0"],
        ["Tensor<[1, 192, 95, 95]> self = ?", "List[int] pad = [2, 2, 2, 2]", "number value = 0.0"],
        ["Tensor<[1, 960, 24, 24]> self = ?", "List[int] pad = [1, 2, 1, 2]", "number value = 0.0"],
```
